src/test/scala/io/shiftleft/fuzzyc2cpg/passes/AstCreationPassTests.scala

```diff
@@ -0,0 +1,734 @@
+    ) { cpg =>
+      cpg.call.name(Operators.delete).code("delete n").argument.code("n").size shouldBe 1
+    }
+
+    "be correct for array 'delete'" in Fixture(
+      """
+        |void delete_number(int n[]) {
+        |  delete[] n;
+        |}
+        |""".stripMargin
+    ) { cpg =>
+      cpg.call.name(Operators.delete).codeExact("delete[] n").argument.code("n").size shouldBe 1
+    }
+
+    "be correct for const_cast" in Fixture(
+      """
+        |void foo() {
+        |  int y = const_cast<int>(n);
+        |  return;
+        |}
+        |""".stripMargin
+    ) { cpg =>
+      cpg.call.name(Operators.cast).codeExact("const_cast<int>(n)").argument.code.l shouldBe List("int", "n")
+    }
+
+    "be correct for static_cast" in Fixture(
+      """
+        |void foo() {
+        |  int y = static_cast<int>(n);
+        |  return;
+        |}
+        |""".stripMargin
+    ) { cpg =>
+      cpg.call.name(Operators.cast).codeExact("static_cast<int>(n)").argument.code.l shouldBe List("int", "n")
+    }
+
+    "be correct for dynamic_cast" in Fixture(
+      """
+        |void foo() {
+        |  int y = dynamic_cast<int>(n);
+        |  return;
+        |}
+        |""".stripMargin
+    ) { cpg =>
+      cpg.call.name(Operators.cast).codeExact("dynamic_cast<int>(n)").argument.code.l shouldBe List("int", "n")
+    }
+
+    "be correct for reinterpret_cast" in Fixture(
+      """
+        |void foo() {
+        |  int y = reinterpret_cast<int>(n);
+        |  return;
+        |}
+        |""".stripMargin
+    ) { cpg =>
+      cpg.call.name(Operators.cast).codeExact("reinterpret_cast<int>(n)").argument.code.l shouldBe List("int", "n")
+    }
+  }
+
+  "AST" should {
+    "have correct line number for method content" in Fixture("""
+                                                                   |
+                                                                   |
+                                                                   |
+                                                                   |
+                                                                   | void method(int x) {
+                                                                   |
+                                                                   |   x = 1;
+                                                                   | }
+      """.stripMargin) { cpg =>
+      cpg.method.name("method").lineNumber.l shouldBe List(6)
+      cpg.method.name("method").block.assignments.lineNumber.l shouldBe List(8)
+    }
+  }
+
+}
+
+object Fixture {
+  def apply(file1Code: String, file2Code: String = "")(f: Cpg => Unit): Unit = {
+    File.usingTemporaryDirectory("fuzzyctest") { dir =>
+      val file1 = (dir / "file1.c")
+      val file2 = (dir / "file2.c")
+      file1.write(file1Code)
+      file2.write(file2Code)
+
+      val cpg = Cpg.emptyCpg
+      val keyPools = Iterator(new IntervalKeyPool(1001, 2000), new IntervalKeyPool(2001, 3000))
```

Comment:
I suggest that at least for the parallel running stuff we encapsulate the key pool creation in AstCreationPass.
This would have the benefit that we have a clearer relation between the individual key pools and the corresponding file names. At the moment the are only associated because they come out of their respective iterators at the same position which is brittle. If we would only pass the id range available for all file names and let the AstCreationPass than create the individual chunks, this would be much clearer.

Reply:
We can't encapsulate the keypool creation in a pass because the individual pass does not know which keys are used by other passes. Keypools need to be assigned to passes by an external entity that keeps track of the existing keypools.

Reply:
I suggested that the AstCreationPass would get one ID range which it than splits up as needed.